From 22ddd73084c7251338762c1745c431c36a37b368 Mon Sep 17 00:00:00 2001
From: Pavla Kratochvilova <pkratoch@redhat.com>
Date: Wed, 5 Dec 2018 09:04:14 +0100
Subject: [PATCH] [transaction] Make transaction content available for commands

---
 dnf/base.py | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/dnf/base.py b/dnf/base.py
index 58bf3a8..0654dde 100644
--- a/dnf/base.py
+++ b/dnf/base.py
@@ -863,12 +863,6 @@ class Base(object):
 #        if self.history.group_active() and self._trans_success:
 #            self.history.group.commit()
 
-        # Plugins were executed already.
-        # They may have used information about the last transaction.
-        # Closing history/transaction here because nobody is supposed to read it anymore.
-        self.history.close()
-        self._history = None
-
         return tid
 
     def _trans_error_summary(self, errstring):
--
libgit2 0.27.7

