From d78d79f44be227a910ec8ecf811ff5898b14f653 Mon Sep 17 00:00:00 2001
From: Jan Kolarik <jkolarik@redhat.com>
Date: Wed, 20 Sep 2023 10:25:59 +0200
Subject: [PATCH] Revert "Block signals during RPM transaction processing"

---
 dnf/base.py | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/dnf/base.py b/dnf/base.py
index ff7c6fe7..7e97e212 100644
--- a/dnf/base.py
+++ b/dnf/base.py
@@ -1028,9 +1028,6 @@ class Base(object):
                 for display_ in cb.displays:
                     display_.output = False
 
-            # block signals to disallow external interruptions of the transaction
-            rpm.blockSignals(True)
-
             self._plugins.run_pre_transaction()
 
             logger.info(_('Running transaction'))
@@ -1048,9 +1045,6 @@ class Base(object):
         for msg in dnf.util._post_transaction_output(self, self.transaction, _pto_callback):
             logger.debug(msg)
 
-        # unblock signals as we are done with the transaction
-        rpm.blockSignals(False)
-
         return tid
 
     def _trans_error_summary(self, errstring):
-- 
2.41.0

