From 2de14efc0c535f10691ff1738caa8c76661e12f3 Mon Sep 17 00:00:00 2001
From: Jaroslav Mracek <jmracek@redhat.com>
Date: Fri, 5 Oct 2018 09:38:30 +0200
Subject: [PATCH 2/2] Fix problem with upgrade in dnf

---
 dnf/base.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dnf/base.py b/dnf/base.py
index 3ec3fba6..b2628150 100644
--- a/dnf/base.py
+++ b/dnf/base.py
@@ -1923,7 +1923,8 @@ class Base(object):
         installed = self.sack.query().installed()
         q = query.intersection(self.sack.query().filterm(name=[pkg.name for pkg in installed]))
         if obsoletes:
-            obsoletes = self.sack.query().available().filterm(obsoletes=q.installed.union(q.upgrades()))
+            obsoletes = self.sack.query().available().filterm(
+                obsoletes=q.installed().union(q.upgrades()))
             # add obsoletes into transaction
             q = q.union(obsoletes)
         # provide only available packages to solver otherwise selection of available
-- 
2.19.0

