diff -uNr dnf-git-40.c0f7fc6/dnf/modules.py dnf-git-40.c0f7fc6.profile/dnf/modules.py
--- dnf-git-40.c0f7fc6/dnf/modules.py	2017-07-17 20:31:42.097392711 +0200
+++ dnf-git-40.c0f7fc6.profile/dnf/modules.py	2017-07-17 20:31:23.710236160 +0200
@@ -251,8 +251,6 @@
             if not stream:
                 # TODO: read default stream from system profile
                 stream = "f26"
-                if name == "httpd":
-                    stream = "2.4"
 
             if not stream:
                 return None
@@ -315,7 +313,7 @@
                 # TODO: nsvap depends on enabled module :(
                 self.enable(nsvap.name, True)
 
-            module_version.install(nsvap.profile)
+            module_version.install(nsvap.profile or "default")
 
     def upgrade(self, pkg_specs):
         for pkg_spec in pkg_specs:
